modules/Bio/EnsEMBL/Compara/RunnableDB/OrthologQM/FlagHighConfidenceOrthologs.pm

```diff
@@ -173,4 +209,57 @@ sub _write_distribution {
     }
 }
 
+sub check_homology_counts {
+    my $self = shift;
+
+    my $homology_file = $self->param_required('homology_file');
+    my $wc_hom = $self->_lines_in_file($homology_file);
+    return (0,0,0) if $wc_hom == 0;
+
+    my $wc_goc = $self->_lines_in_file($self->param('goc_file'));
+    my $wc_wga = $self->_lines_in_file($self->param('wga_file'));
+    return ($wc_hom, $wc_goc, $wc_wga);
+}
+
+sub _lines_in_file {
+    my ( $self, $filename ) = @_;
+
+    my @wc_out = split( /\s+/, $self->get_command_output("wc -l $filename") );
+    my $wc_l   = shift @wc_out;
+    return $wc_l - 1; # account for header line
+}
```

Comment:
I'm conflicted about whether to change to this solution or not. If I'm understanding `sysread` correctly, it would read the whole file into memory? My concern is similar to that with the use of `_spurt` - should we be reading these potentially large files into memory when there are lighter alternatives available (and already implemented 😉 )

Reply:
I much prefer the original solution (running `wc -l`). It's very clear and I don't think it will be a performance bottleneck